### PR TITLE
CA-164, 165 - Added descriptive video link functionality

### PR DESCRIPTION
**Pull request by marcbeermann**
- can now add a descriptive YouTube video link on create project page
- can edit the descriptive video link on the edit project page
- can view descriptive YouTube video on the display page
- ApplicationDbContext.cs -> added 'on delete cascade' behaviour so that on project deletion related videoLink items are also deleted

### CODE REVIEW

File: src/CollAction/Controllers/ProjectsController.cs [hunk 1]
```diff
@@ -204,6 +212,28 @@ public ViewResult StartInfo()
                 return View(editProjectViewModel);
             }
 
+            VideoLink descriptionVideoLink = null;
+
+            // If the video link hasn't changed...
+            if (project.DescriptionVideoLink?.Link == editProjectViewModel.DescriptionVideoLink)
```
Comment (stewartmatheson) [inline src/CollAction/Controllers/ProjectsController.cs]: Could this be extracted somewhere else other than the controller? It seems like this logic should belong to a model layer somewhere.

File: src/CollAction/Controllers/ProjectsController.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ public ViewResult StartInfo()
         [HttpPost]
         [ValidateAntiForgeryToken]
         [Authorize]
-        public async Task<IActionResult> Create([Bind("Name,Description,Goal,Proposal,CreatorComments,CategoryId,LocationId,Target,End")] CreateProjectViewModel createProjectViewModel)
+        public async Task<IActionResult> Create([Bind("Name,Description,Goal,Proposal,CreatorComments,CategoryId,LocationId,Target,End,DescriptionVideoLink")] CreateProjectViewModel createProjectViewModel)
```
Comment (logophobia) [inline src/CollAction/Controllers/ProjectsController.cs]: It's fine to not use bind, but you don't "have" to use it if you have an appropiately restricted viewmodel. Might be a bit easier, just as long as you're sure you can handle a viewmodel with all parameters filled in.

File: src/CollAction/Models/ProjectViewModels/DisplayProjectViewModel.cs [hunk 3]
```diff
@@ -15,6 +16,21 @@ public class DisplayProjectViewModel
 
         public int Participants { get; set; }
 
+        public bool HasDescriptionVideo { get { return Project.DescriptionVideoLink != null; } }
+
+        public string DescriptionVideoYouTubeEmbedLink {
+            get {
+                if (!HasDescriptionVideo) { return ""; }
+
+                var regex = new Regex(@"^(?:https?:\/\/www\.youtube\.com\/watch\?v=)((?:\w|-){11,})(?:\S+)?$");
```
Comment (logophobia) [inline src/CollAction/Models/ProjectViewModels/DisplayProjectViewModel.cs]: Prehaps extract this to a validation attribute? Also, might better to use a url parser for stuff like this, a regex isn't the best way to parse urls.

**Comment by logophobia** [comment]
Some small things, but looks good.

File: src/CollAction/Controllers/ProjectsController.cs [hunk 4]
```diff
@@ -108,6 +108,12 @@ public ViewResult StartInfo()
                 return View(createProjectViewModel);
             }
 
+            VideoLink descriptionVideoLink = null;
+            if (createProjectViewModel.DescriptionVideoLink != null)
+            {
+                descriptionVideoLink = new VideoLink { Link = createProjectViewModel.DescriptionVideoLink, Date = DateTime.UtcNow };
+            }
```
Comment (logophobia) [inline src/CollAction/Controllers/ProjectsController.cs]: Small style point (so feel free to ignore, mostly taste). You could do:
VideoLink descriptionVideoLink = 
    createProjectViewModel.DescriptionVideoLink == null ?
        null :
        new VideoLink { Link = createProjectViewModel.DescriptionVideoLink, Date = DateTime.UtcNow };

**Comment by logophobia** [comment]
Looks good, feel free to merge

**Pull request merged**